### PR TITLE
Added timestamp, altitude & precision loc fields

### PR DESCRIPTION
**Pull request by Davo36**
Menno, I think I've managed to make the first set of changes here i.e. to management-interface.

Not 100% sure it's the .yaml format you want, but a bit tricky to change that.  Have to do custom marshalling functions I think.

### CODE REVIEW

**Comment by mjs** [comment]
It turns out that much of this is unnecessary. `time.Time` implements the `encoding.TextMarshaler interface` and `encoding.TextUnmarshaler` interfaces which is used by the yaml package to serialise time values using the format we want (RFC3339 which is similar to ISO 8601).

Example: https://gist.github.com/mjs/6dbdeaea8bd2e0df0abc295b6b1f3897

Sorry for not pointing you in the right direction sooner.

**Comment by Davo36** [comment]
Menno, you use that raw struct for all the other values as well. I assume this was your code or was it someone elses?

Any way, I can remove that struct, but it's a complete rewrite of all that stuff.  Is that what you want?

File: helpers.go [hunk 1]
```diff
@@ -44,6 +45,22 @@ func floatToString(val float64) string {
...
 }
 
+// Timestamp returns the field at 'val' as a time value
+func parseTimestamp(val string) (time.Time, bool) {
+	if val == "" {
+		return time.Now(), true
+	}
+	t, err := time.Parse(time.RFC3339, val)
+	if err != nil {
+		return time.Now(), false
```
Comment (mjs) [inline helpers.go]: It's fine to `return time.Time{}, false` in the error case

File: location.go [hunk 2]
```diff
@@ -87,6 +88,7 @@ func handleLocationPostRequest(w http.ResponseWriter, r *http.Request) (*rawLoca
 	} else {
 		var err error
 		rawLocation = newRawLocationData(r)
+		rawLocation.Timestamp = time.Now().Format(time.RFC3339) // Update the timestamp.
```
Comment (mjs) [inline location.go]: use `timestampToString` function here?

**Comment by mjs** [comment]
Ah, I remember now. The `rawLocationData` struct is used for the web form so that we can preserve invalid input so that the user can correct what they typed.

In that case, what you have is fine. Reviewing again now.

**Pull request merged**